modules/parent-join/src/main/java/org/elasticsearch/join/aggregations/ParentToChildrenAggregator.java

```diff
@@ -105,17 +107,16 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
         if (valuesSource == null) {
             return LeafBucketCollector.NO_OP_COLLECTOR;
         }
-
-        final SortedDocValues globalOrdinals = valuesSource.globalOrdinalsValues(parentType, ctx);
-        assert globalOrdinals != null;
+        final SortedSetDocValues globalOrdinals = valuesSource.globalOrdinalsValues(ctx);
```

Comment:
It would be possible only when the shard has a single segment otherwise the returned object is a GlobalOrdinalMapping and not a SingletonSortedSetDocValues. We could have a specialized SingletonGlobalOrdinalMapping but maybe in a follow up ?